public/app/features/alerting/unified/AmRoutes.tsx

```diff
@@ -66,23 +68,50 @@ const AmRoutes: FC = () => {
       exitRootRouteEditMode();
     }
 
-    console.log(newData);
-
-    fetchConfig();
+    dispatch(
+      updateAlertManagerConfigAction({
+        newConfig: {
+          ...result,
+          alertmanager_config: {
+            ...result.alertmanager_config,
+            route: newData,
+          },
+        },
+        oldConfig: result,
+        alertManagerSourceName: alertManagerSourceName!,
+        successMessage: 'Saved',
+      })
+    );
   };
 
+  useEffect(() => {
+    if (savingDispatched && !saving) {
+      fetchConfig();
+    }
+  }, [fetchConfig, savingDispatched, saving]);
+
+  if (!alertManagerSourceName) {
+    return <Redirect to="/alerting/routes" />;
+  }
+
   return (
     <AlertingPageWrapper pageId="am-routes">
       <Field label="Choose alert manager">
         <AlertManagerPicker current={alertManagerSourceName} onChange={setAlertManagerSourceName} />
       </Field>
-      {error && !loading && (
+      {savingError && !saving && (
+        <Alert severity="error" title="Error saving alert manager config">
+          {savingError.message || 'Unknown error.'}
+        </Alert>
+      )}
+      {resultError && !resultLoading && (
         <Alert severity="error" title="Error loading alert manager config">
-          {error.message || 'Unknown error.'}
+          {resultError.message || 'Unknown error.'}
         </Alert>
       )}
-      {loading && <LoadingPlaceholder text="loading alert manager config..." />}
-      {result && !loading && !error && (
+      {resultLoading && <LoadingPlaceholder text="loading alert manager config..." />}
+      {saving && <LoadingPlaceholder text="saving alert manager config..." />}
+      {result && !resultLoading && !saving && !resultError && (
```

Comment:
imho if there's old result, we can show it until new result loads? to prevent sudden flashing of loader